src/Jobs/DeleteTenantsPostgresUser.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\DB;
+use Stancl\Tenancy\Database\Contracts\TenantWithDatabase;
+
+class DeleteTenantsPostgresUser implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(
+        protected TenantWithDatabase&Model $tenant,
+    ) {
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $name = $this->tenant->getTenantKey();
+
+        // Revoke all permissions of a Postgres user before dropping it
+        // Skip dropping permissions if the user doesn't exist
+        if (count(DB::select("SELECT usename FROM pg_user WHERE usename = '$name';")) > 0) {
+            DB::statement("DROP OWNED BY \"{$name}\";");
+            DB::statement("DROP USER \"{$name}\";");
+        }
```

Comment:
Can we do things like this in a transaction? If yes, then we should probably use transactions. Same in other parts of the PR.